xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -1029,3 +1060,132 @@ DXGI_COLOR_SPACE_TYPE CProcessorHD::AvToDxgiColorSpace(const DXGIColorSpaceArgs&
 
   return DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P709;
 }
+
+bool CProcessorHD::IsSuperResolutionSuitable(const VideoPicture& picture)
+{
+  if (picture.iWidth > 1920)
+    return false;
+
+  const UINT outputWidth = DX::Windowing()->GetBackBuffer().GetWidth();
+
+  if (outputWidth <= picture.iWidth)
+    return false;
+
+  if (picture.iFlags & DVP_FLAG_INTERLACED)
+    return false;
+
+  if (picture.color_primaries == AVCOL_PRI_BT2020 ||
+      picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
+      picture.color_transfer == AVCOL_TRC_ARIB_STD_B67)
+    return false;
+
+  // Big memory leak with SDR x264 10-bit videos
+  // ToDo: retest with future NVIDIA drivers
+  if (picture.colorBits == 10)
```

Comment:
Latest NVIDIA driver 535.98 has fixed mem leak in h.264/AVC 10 bit, then this can be removed.

Vertical shift is not fixed yet.

Reply:
That's good news, SDR 10 bit is artificial precision (AVC or HEVC) and throwing away 2 bits would have a marginal effect, Would there be a benefit in making the decoder output 8 bits in order to make it compatible with super resolution and take advantage of the better upscaler?

Reply:
RTX Super Resolution is ACTIVE and works with x264 10 bit using NVIDIA driver 535.98